### PR TITLE
add Typescript type definitions.

### PR DESCRIPTION
**Pull request by jason-chang**
add Typescript type definitions.

### CODE REVIEW

File: types.d.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    type UANameKey = 'name';
+    type UAVersionKey = 'version';
+
+    type UAParserExtensions = Array<RegExp[] | [UANameKey | [UANameKey, string], UAVersionKey]>;
+
+    type UABrowserResult = { name: string, version: string };
+    type UADeviceResult = { model: string, type: string, vendor: string };
+    type UAEngineResult = { name: string, version: string };
+    type UAOSResult = { name: string, version: string };
+    type UACpuResult = { architecture: string };
```
Comment (artola) [inline types.d.ts:38]: case as the method `getCPU ` ==> `UACPUResult`

**Comment by qkreltms** [comment]
[@types/ua-parser-js](https://www.npmjs.com/package/@types/ua-parser-js)

**Comment by jason-chang** [comment]
@qkreltms 👌 ，got it，i'll close this。

**Comment by jason-chang** [comment]
[@types/ua-parser-js](https://www.npmjs.com/package/@types/ua-parser-js) so close this pr.

**Pull request closed without merging**